webforj-plugins/webforj-install-bbj-plugin/src/main/java/com/webforj/installer/WebforjInstaller.java

```diff
@@ -119,59 +112,68 @@ private String unzipPom(String zipFile) throws IOException {
   private void unzipBbjProgs(String zipFile, String directory) throws IOException {
     log.info("uzipBbjProgs: zipFile = {}, directory = {}", zipFile, directory);
     File file = new File(zipFile);
-    try (ZipFile zip = new ZipFile(file)) {
+    // Ensure valid path ends with system separator character for zip slip issues.
+    String targetDirectory = StringUtils.endsWithIgnoreCase(directory, File.separator) ? directory
+        : directory.concat(File.separator);
+    log.info("directory = {}, targetDirectory = {}", directory, targetDirectory);
 
+    try (ZipFile zip = new ZipFile(file)) {
       Enumeration<? extends ZipEntry> zipFileEntries = zip.entries();
-
-      log.info("processing entries");
+      log.info("processing zip entries ...");
       // Process each entry
       while (zipFileEntries.hasMoreElements()) {
         // grab a zip file entry
         ZipEntry entry = zipFileEntries.nextElement();
 
-        String currentEntry = entry.getName();
-        if (!currentEntry.endsWith(".bbj")) {
+        String currentEntry = FilenameUtils.normalize(entry.getName());
+        if (!"bbj".equalsIgnoreCase(FilenameUtils.getExtension(currentEntry))) {
           continue;
         }
 
-        File destFile = new File(directory, entry.getName());
-        log.info("extracting {}", destFile.getAbsoluteFile());
-        out.append("webforj-installer: extracting ").append(destFile.getAbsolutePath())
-            .append("\n");
-        Files.deleteIfExists(destFile.toPath());
-        Paths.get(destFile.getPath()).getParent().toFile().mkdirs();
-
-        log.info("writing current file {} to disk", destFile);
-        zipBufferedInputStream(zip, entry, destFile);
+        File destFile = new File(targetDirectory + currentEntry);
+        String canonicalDestinationPath = destFile.getCanonicalPath();
+        if (canonicalDestinationPath.startsWith(targetDirectory)) {
+          log.info("extracting {}", canonicalDestinationPath);
+          out.append("webforj-installer: extracting %s%n".formatted(canonicalDestinationPath));
+          Files.deleteIfExists(destFile.toPath());
+          File parent = destFile.getParentFile();
+          log.info("testing parent directory {}, attempting to create if not exists", parent);
+          if (!parent.exists() && !parent.mkdirs()) {
```

Comment:
https://sonarcloud.io/project/issues?id=DwcJava_engine&issues=AY4y0kXmf_KH3f9oPyJK&open=AY4y0kXmf_KH3f9oPyJK

Reply:
This is the fix that sonar describes as the solution to the zip slip:
```

        File destFile = new File(targetDirectory + currentEntry);
        String canonicalDestinationPath = destFile.getCanonicalPath();
        if (canonicalDestinationPath.startsWith(targetDirectory)) {
```
...